File: 0344-reverse-string/solution.cpp
// 344. Reverse String
// https://leetcode.com/problems/reverse-string/
// Difficulty: Easy | Language: cpp
// Runtime: 11 ms
// Memory: 27.2 MB
// Solved: 2024-07-11

class Solution {
public:
    void reverseString(vector<char>& s) {
        int start = 0; 
        int end = s.size()-1;
        while(start< end){
            swap(s[start++] , s[end--]);

        }
    
        
    }
};

File: 0344-reverse-string/solution.py
# 344. Reverse String
# https://leetcode.com/problems/reverse-string/
# Difficulty: Easy | Language: python3
# Runtime: 7 ms
# Memory: 22.9 MB
# Solved: 2025-05-02

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        for i in range(0,len(s)//2):
            s[(len(s)-1)-i],s[i] = s[i], s[(len(s)-1)-i]
